sparse_strips/vello_common/src/encode.rs

```diff
@@ -520,6 +528,24 @@ pub struct EncodedImage {
     pub y_advance: Vec2,
 }
 
+impl EncodedImage {
+    /// Sample a premultiplied RGBA8 pixel from the encoded image.
+    #[inline(always)]
+    pub fn sample(&self, x: u16, y: u16) -> PremulRgba8 {
+        let idx = 4 * (self.width as usize * y as usize + x as usize);
+        let data = &self.blob.data()[idx..][..4];
+
+        let global_alpha = (self.alpha * 255. + 0.5) as u8;
+        let alpha = ((data[3] as u16 * global_alpha as u16) / 255) as u8;
+        PremulRgba8 {
+            r: ((data[0] as u16 * alpha as u16) / 255) as u8,
+            g: ((data[1] as u16 * alpha as u16) / 255) as u8,
+            b: ((data[2] as u16 * alpha as u16) / 255) as u8,
+            a: alpha,
+        }
+    }
```

Comment:
This might not be the best place for doing this conversion, and it will get more complicated if `peniko::ImageFormat` supports more than `Rgba8` in the future.